### PR TITLE
Enable wayland

### PR DESCRIPTION
**Pull request by mariogrip**
This enables wayland to work correctly

### CODE REVIEW

File: src/plugin/inputmethod_p.h [hunk 1]
```diff
@@ -177,12 +177,11 @@ class InputMethodPrivate
 
         if (QGuiApplication::platformName() == "ubuntumirclient") {
             view->setFlags(InputMethodWindowType); /* Mir-only OSK window type */
```
Comment (NeoTheThird) [inline src/plugin/inputmethod_p.h:179]: Do we have a convention for single-line comments? `//` or `/* */`? Or do we just not care?

Comment (dobey) [reply src/plugin/inputmethod_p.h:179]: I think most of this component uses `//` and plenty of multi-line comments use that too. But also this code is pre-existing and not part of this change set, so we shouldn't change it here.

**Review by dobey — APPROVED** [review]

File: src/plugin/inputmethod_p.h (hunk 1 above)
Comment (NeoTheThird) [reply src/plugin/inputmethod_p.h:179]: right

**Comment by NeoTheThird** [comment]
@mariogrip you decide when to merge

**Pull request merged**